tubular/numeric.py

```diff
@@ -518,3 +525,190 @@ def transform(self, X):
             ].product(axis=1, skipna=False)
 
         return X
+
+
+class PCATransformer(BaseTransformer):
+    """Transformer that generates variables using Principal component analysis (PCA).
+    Linear dimensionality reduction using Singular Value Decomposition of the
+    data to project it to a lower dimensional space.
+
+    It is based on sklearn class sklearn.decomposition.PCA
+
+        Parameters
+        ----------
+        columns : None or list or str
+            Columns to apply the transformer to. If a str is passed this is put into a list. Value passed
+            in columns is saved in the columns attribute on the object. Note this has no default value so
+            the user has to specify the columns when initialising the transformer. This is avoid likely
+            when the user forget to set columns, in this case all columns would be picked up when super
+            transform runs.
+        n_components : int, float or 'mle', default=None
+            Number of components to keep.
+            if n_components is not set all components are kept::
+                n_components == min(n_samples, n_features)
+            If ``n_components == 'mle'`` and ``svd_solver == 'full'``, Minka's
+            MLE is used to guess the dimension. Use of ``n_components == 'mle'``
+            will interpret ``svd_solver == 'auto'`` as ``svd_solver == 'full'``.
+            If ``0 < n_components < 1`` and ``svd_solver == 'full'``, select the
```

Comment:
Is it possible for us to check this and give an error if condition not met? if not then a warning may suffice

Reply:
Commit : [svd_sovler and float condition](https://github.com/lvgig/tubular/pull/57/commits/22622e93cbaf091bf9d546af9ad3d85f6f7bd763)